From: Hannes Draeger <hdraeger@phoenixcontact.com>
Date: Tue, 27 Mar 2018 15:35:33 +0100
Subject: PAM - add hardened options to password rules

This patch adds the pam_cracklib.so to the common-password rules

---  a/pam.d/common-password	2018-03-28 10:12:45.458180169 +0200
+++  b/pam.d/common-password-changes	2018-03-28 15:20:27.954966644 +0200
@@ -15,8 +15,11 @@
 #
 # See the pam_unix manpage for other options.
 
+#addition of pam_cracklib to get a password length and further checks
+password requisite pam_cracklib.so retry=3 minlen=8 enforce_for_root
+
 # here are the per-package modules (the "Primary" block)
-password	[success=1 default=ignore]	pam_unix.so obscure sha512
+password	[success=1 default=ignore]	pam_unix.so obscure sha512 shadow use_authtok
 # here's the fallback if no module succeeds
 password	requisite			pam_deny.so
 # prime the stack with a positive return value if there isn't one already;
